packages/messaging/src/controllers/window-controller.ts

```diff
@@ -308,16 +308,23 @@ export class WindowController extends BaseController {
           // This message has a campaign id, meaning it was sent using the FN Console.
           // Analytics is enabled on this message, so we should log it.
           const eventType = getEventType(firebaseMessagingType);
-          (this.app as _FirebaseApp).INTERNAL.analytics.logEvent(
-            eventType,
-            /* eslint-disable camelcase */
-            {
-              message_name: data[FN_CAMPAIGN_NAME],
-              message_id: data[FN_CAMPAIGN_ID],
-              message_time: data[FN_CAMPAIGN_TIME],
-              message_device_time: Math.floor(Date.now() / 1000)
+          this.services.analyticsProvider.get().then(
```

Comment:
Can you make the event listener `async` and use `await` here?

Reply:
Done.